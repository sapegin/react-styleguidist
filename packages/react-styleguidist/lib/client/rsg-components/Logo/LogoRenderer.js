import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';

const styles = function styles(_ref) {
  const color = _ref.color;

      
const fontFamily = _ref.fontFamily;

      
const fontSize = _ref.fontSize;
  return {
    logo: {
      color: color.base,
      margin: 0,
      fontFamily: fontFamily.base,
      fontSize: fontSize.h4,
      fontWeight: 'normal'
    }
  };
};

export function LogoRenderer(_ref2) {
  const classes = _ref2.classes;

      
const children = _ref2.children;
  return React.createElement("h1", {
    className: classes.logo
  }, children);
}
LogoRenderer.propTypes = {
  classes: PropTypes.object.isRequired,
  children: PropTypes.node
};
export default Styled(styles)(LogoRenderer);