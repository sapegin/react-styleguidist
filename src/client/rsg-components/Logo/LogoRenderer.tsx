import React from 'react';
import Styled, { JssInjectedProps } from 'rsg-components/Styled';
import { ProcessedTheme } from '../../../typings/Theme';

const styles = ({ color, fontFamily, fontSize }: ProcessedTheme) => ({
	logo: {
		color: color.base,
		margin: 0,
		fontFamily: fontFamily.base,
		fontSize: fontSize.h4,
		fontWeight: 'normal',
	},
});

export const LogoRenderer: React.FunctionComponent<JssInjectedProps> = ({ classes, children }) => {
	return <h1 className={classes.logo}>{children}</h1>;
};

export default Styled(styles)(LogoRenderer);
