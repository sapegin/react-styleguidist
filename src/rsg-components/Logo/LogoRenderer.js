import React from 'react';
import PropTypes from 'prop-types';
import Heading from 'rsg-components/Heading';
import Styled from 'rsg-components/Styled';

const styles = ({ fontSize }) => ({
	logoText: {
		fontSize: fontSize.h4,
	},
});

function LogoRenderer({ classes, children }) {
	return (
		<Heading level={1}>
			<div className={classes.logoText}>{children}</div>
		</Heading>
	);
}

LogoRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	children: PropTypes.node,
};

export default Styled(styles)(LogoRenderer);
