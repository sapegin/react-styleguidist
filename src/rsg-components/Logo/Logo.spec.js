import React from 'react';
import LogoRenderer from './LogoRenderer';

it('renderer should render header', () => {
	const actual = shallow(<LogoRenderer>React Styleguidist</LogoRenderer>);

	expect(actual.dive()).toMatchSnapshot();
});
