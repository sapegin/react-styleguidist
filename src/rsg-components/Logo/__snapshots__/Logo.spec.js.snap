// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render header 1`] = `
<Styled(Heading)
  level={1}
>
  <div
    className="rsg--logoText-0"
  >
    React Styleguidist
  </div>
</Styled(Heading)>
`;
